File: backend/src/main/kotlin/com/github/plplmax/planning/timeslots/Timeslot.kt
@file:UseSerializers(LocalTimeSerializer::class, DayOfWeekSerializer::class)

package com.github.plplmax.planning.timeslots

import ai.timefold.solver.core.api.domain.lookup.PlanningId
import com.github.plplmax.planning.plugins.serialization.DayOfWeekSerializer
import com.github.plplmax.planning.plugins.serialization.LocalTimeSerializer
import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import java.time.DayOfWeek
import java.time.LocalTime

@Serializable
data class Timeslot(@PlanningId val id: Int, val dayOfWeek: DayOfWeek, val start: LocalTime, val end: LocalTime)

File: frontend/src/timeslots/Timeslot.ts
import type { LocalTime } from '@js-joda/core'

export interface Timeslot {
  id: number
  dayOfWeek: number
  start: LocalTime
  end: LocalTime
}

export interface NewTimeslot {
  dayOfWeek: number
  start: LocalTime
  end: LocalTime
}

export interface DayOfWeek {
  id: number
  label: string
  shortLabel: string
}

export const dayOfWeeks: DayOfWeek[] = [
  {
    id: 1,
    label: 'Понедельник',
    shortLabel: 'Пн'
  },
  {
    id: 2,
    label: 'Вторник',
    shortLabel: 'Вт'
  },
  {
    id: 3,
    label: 'Среда',
    shortLabel: 'Ср'
  },
  {
    id: 4,
    label: 'Четверг',
    shortLabel: 'Чт'
  },
  {
    id: 5,
    label: 'Пятница',
    shortLabel: 'Пт'
  },
  {
    id: 6,
    label: 'Суббота',
    shortLabel: 'Сб'
  },
  {
    id: 7,
    label: 'Воскресенье',
    shortLabel: 'Вс'
  }
]
